controller/controller.go

```diff
@@ -125,6 +125,21 @@ func wrapDBExec(dbExec func(string, ...interface{}) error) func(string, ...inter
 }
 
 func streamRouterEvents(rc routerc.Client, db *postgres.DB, doneCh chan struct{}) error {
+	// wait for router to come up
+	{
+		events := make(chan *discoverd.Event)
+		stream, err := discoverd.NewService("router-api").Watch(events)
+		if err != nil {
+			return err
+		}
+		for e := range events {
```

Comment:
I know this is merged, but we should probably return an error if `events` gets closed (rather than panicking as it will do now).